solr/core/src/java/org/apache/solr/schema/ScalarQuantizedDenseVectorField.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.schema;
+
+import static java.util.Optional.ofNullable;
+import static org.apache.lucene.codecs.lucene99.Lucene99HnswVectorsFormat.DEFAULT_NUM_MERGE_WORKER;
+
+import java.util.Map;
+import org.apache.lucene.codecs.KnnVectorsFormat;
+import org.apache.lucene.codecs.lucene99.Lucene99HnswScalarQuantizedVectorsFormat;
+import org.apache.lucene.codecs.lucene99.Lucene99ScalarQuantizedVectorsFormat;
+import org.apache.lucene.index.VectorEncoding;
+import org.apache.lucene.index.VectorSimilarityFunction;
+import org.apache.solr.common.SolrException;
+
+public class ScalarQuantizedDenseVectorField extends DenseVectorField {
+  public static final String BITS_PARAM = "bits"; //
+  public static final String CONFIDENCE_INTERVAL_PARAM = "confidenceInterval";
+  public static final String DYNAMIC_CONFIDENCE_INTERVAL_PARAM = "dynamicConfidenceInterval";
+  public static final String COMPRESS_PARAM =
+      "compress"; // can only be enabled when bits = 4 per Lucene codec spec
+
+  static final int DEFAULT_BITS = 7; // use signed byte as default when unspecified
+  static final Float DEFAULT_CONFIDENCE_INTERVAL = null; // use dimension scaled confidence interval
+
+  // lucene does not expose these so we are duplicating them here
+  static final Float MINIMUM_CONFIDENCE_INTERVAL = 0.9f;
```

Comment:
I'm not sure we need this here, looking at Lucene code, in case we go off the values we get an "llegalArgumentException(
          "confidenceInterval must be between "
              + MINIMUM_CONFIDENCE_INTERVAL
              + " and "
              + MAXIMUM_CONFIDENCE_INTERVAL
              + " or 0"
              + "; confidenceInterval="
              + confidenceInterval);"
              
              Can't we catch this exception and handle it rather than duplicating the constants?